### PR TITLE
fix: Unwanted match of charset (922100 PL1)

### PR DESCRIPTION
**Pull request by azurit**
Fix for [this](https://github.com/coreruleset/coreruleset/issues/3114#issuecomment-1407350626) issue.

Label `v4` can be removed from issue https://github.com/coreruleset/coreruleset/issues/3114 after this is merged (but do NOT close that issue).

Fixes: #3118

### CODE REVIEW

**Comment by dune73** [comment]
@theseion Would you mind reviewing this to check whether it implements your idea correctly?

File: rules/REQUEST-922-MULTIPART-ATTACK.conf [hunk 1]
```diff
@@ -40,8 +40,9 @@ SecRule &MULTIPART_PART_HEADERS:_charset_ "!@eq 0" \
     tag:'paranoia-level/1',\
     ver:'OWASP_CRS/4.0.0-rc2',\
     severity:'CRITICAL',\
+    setvar:'tx.922100_charset=|%{ARGS._charset_}|',\
```
Comment (M4tteoP) [inline rules/REQUEST-922-MULTIPART-ATTACK.conf:43]: Not sure about it, but would be possible to write a test that fails before the fix and works after?

Comment (azurit) [reply rules/REQUEST-922-MULTIPART-ATTACK.conf:43]: @M4tteoP About exactly what are you not sure?

Comment (theseion) [reply rules/REQUEST-922-MULTIPART-ATTACK.conf:43]: I think he's asking for a test case, which I think is a good idea too. Here it is:

```yaml
  - test_title: 922100-3
    desc: "Negative test: don't allow substrings of valid charsets when using _charset_"
    stages:
      - stage:
          input:
            dest_addr: 127.0.0.1
            headers:
              Host: "localhost"
              User-Agent: "OWASP CRS test agent"
              Content-Type: multipart/form-data; boundary=boundary
              Accept: "*/*"
            method: POST
            port: 80
            uri: "/post"
            version: "HTTP/1.1"
            data: |
              --boundary
              Content-disposition: form-data; name="_charset_"

              utf-
              --boundary
              Content-disposition: form-data; name="negative"
              Content-Type: text/plain

              utf-8 is valid but utf- should not be.
              --boundary--
          output:
            no_log_contains: id "922100"

```

**Review by theseion — CHANGES_REQUESTED** [review]

File: rules/REQUEST-922-MULTIPART-ATTACK.conf (hunk 1 above)
Comment (M4tteoP) [reply rules/REQUEST-922-MULTIPART-ATTACK.conf:43]: Yes, I was asking for a test case. Thanks!

**Comment by azurit** [comment]
Test was added (@theseion thanks).

**Review by theseion — APPROVED** [review]

**Pull request merged**